sql/wsrep_mysqld.cc

```diff
@@ -2283,19 +2290,12 @@ void wsrep_close_client_connections(my_bool wait_to_end, THD* except_caller_thd)
     First signal all threads that it's time to die
   */
 
-  mysql_mutex_lock(&LOCK_thread_count); // For unlink from list
-
   bool kill_cached_threads_saved= kill_cached_threads;
   kill_cached_threads= true; // prevent future threads caching
   mysql_cond_broadcast(&COND_thread_cache); // tell cached threads to die
 
   server_threads.iterate(kill_all_threads, except_caller_thd);
-  mysql_mutex_unlock(&LOCK_thread_count);
-
-  if (thread_count)
-    sleep(2);                               // Give threads time to die
 
-  mysql_mutex_lock(&LOCK_thread_count);
```

Comment:
Here `LOCK_thread_count` is removed, but the code still broadcasts signal to `COND_thread_cache`. Is it protected by any mutex or should the broadcast also be removed?